express-notes-api/index.js

```diff
@@ -0,0 +1,106 @@
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const errorMsg = {
+  error: 'id must be a postive integer'
+};
+const errorMsgEmpty = {
+  error: 'content is a required field'
+};
+const errorUnexpected = {
+  error: 'An unexpected error occurred.'
+};
+app.listen(3000, () => {
+});
+
+fs.readFile('data.json', 'utf-8', (err, txt) => {
+  if (err) throw err;
+  const txtParsed = JSON.parse(txt);
+  const notesArray = [];
+  for (const key in txtParsed.notes) {
+    notesArray.push(txtParsed.notes[key]);
+  }
+  app.get('/api/notes', (req, res) => {
+    res.status(200).send(notesArray);
+  });
+  app.get('/api/notes/:id', (req, res) => {
+    const id = req.params.id;
+    const idString = (req.params.id).toString();
+    if (id <= 0) {
+      res.status(400).send(errorMsg);
+    } else if (txtParsed.notes[idString] === undefined) {
+      const errorMissingMsg = {
+        error: `cannot find note with id ${id}`
+      };
+      res.status(404).send(errorMissingMsg);
+    }
+    const notesSpecific = txtParsed.notes[id];
+    res.status(200).send(notesSpecific);
```

Comment:
There should be an `else` here.

Reply:
Please address this too. Responding doesn't halt the javascript code, so you need an `else`.